File: security_assistant/remediation/code_examples/command_injection_fix.go
// VULNERABLE:
// cmd := exec.Command("sh", "-c", "ping "+userInput)
// output, err := cmd.Output()

// SECURE (Direct command with arguments):
import (
    "os/exec"
    "regexp"
)

// Validate input
if !regexp.MustCompile(`^[a-zA-Z0-9.-]+$`).MatchString(userInput) {
    return errors.New("invalid hostname")
}

// Use direct command (no shell)
cmd := exec.Command("ping", "-c", "4", userInput)
output, err := cmd.Output()

File: security_assistant/remediation/code_examples/command_injection_fix.java
// VULNERABLE:
// Runtime.getRuntime().exec("ping " + userInput);

// SECURE (Array-based command with validation):
import java.util.regex.Pattern;

// Validate input
if (!Pattern.matches("^[a-zA-Z0-9.-]+$", userInput)) {
    throw new IllegalArgumentException("Invalid hostname");
}

// Use array form (no shell interpretation)
ProcessBuilder pb = new ProcessBuilder("ping", "-c", "4", userInput);
Process process = pb.start();
